packages/flutter/lib/src/painting/geometry.dart

```diff
@@ -48,27 +48,19 @@ Offset positionDependentBox({
   // VERTICAL DIRECTION
   final bool fitsBelow = target.dy + verticalOffset + childSize.height <= size.height - margin;
   final bool fitsAbove = target.dy - verticalOffset - childSize.height >= margin;
-  final bool tooltipBelow = preferBelow ? fitsBelow || !fitsAbove : !(fitsAbove || !fitsBelow);
+  final bool tooltipBelow = fitsAbove == fitsBelow ? preferBelow : fitsBelow;
```

Comment:
Are these changes supposed to be in this PR? Just asking, since the PR description didn't mention it.

Reply:
Yeah drive-through changes, should be covered by existing tests.